src/DOMNodeComparator.php

```diff
@@ -119,10 +119,13 @@ public function assertEquals($expected, $actual, $delta = 0.0, $canonicalize = f
     private function nodeToText(DOMNode $node, $canonicalize, $ignoreCase)
     {
         if ($canonicalize) {
-            $document = new DOMDocument;
-            $document->loadXML($node->C14N());
+            $canonicalizedNode = (string) $node->C14N();
+            if ($canonicalizedNode !== "") {
```

Comment:
Probably don't need the explicit cast:

~~~php
if ($canonicalizedNode = $node->C14N()) {
~~~